pkg/proxy/accesslog/record.go

```diff
@@ -195,3 +198,23 @@ type LogRecordHTTP struct {
 	// Header is the HTTP header in use
 	Header http.Header
 }
+
+// LogRecordKafka contains the kafka specific portion of a log record
+type LogRecordKafka struct {
+	// APIVersion of the kafka api used
+	APIVersion int
+
+	// APIKey for Kafka message
+	// Reference: https://kafka.apache.org/protocol#protocol_api_keys
+	APIKey int16
+
+	// CorrelationID is a user-supplied integer value that will be passed
+	// back with the response
+	CorrelationID int32
+
+	// Topics of the request, can be a single topic for message type
+	// produce/createTopic or a list of topics (fetch/deleteTopic).
+	// Note that this list can be empty since not all messages use
+	// Topic. example: LeaveGroup, Heartbeat
+	Topics []string `json:"Topics,omitempty"`
```

Comment:
I would consider making this `Topics []Topic` with `Topic` being a struct to make it extendable without breaking parsers if we ever want to report things like error codes in responses.

Reply:
Error codes may or may not be related to Topic. Topic is not a mandatory field in other message types like JoinGroup, LeaveGroup, HeartBeat et al.  The reason why error code was not added as part of LogRecordKafka right now, if because for some responses like Fetch, there can be a list of error codes returned per topic. https://kafka.apache.org/protocol#The_Messages_Fetch

Reply:
Let me know if you still want to include error in Topic structure, just targeting messaging which deal with Topics. I can change Topic string --> structure for sure.

Reply:
Done. I have been non-creative with field names, more than welcome to change the name.

Reply:
I don't think we have to include error right now but I want to prepare to include topic-specific fields later on without breaking existing parsers, whether it is error codes or something else. A struct is extendable, a string is not.

Reply:
agreed.Made it into a struct.